### DISCUSSION TITLE
max number of member fields in a structure

### DISCUSSION THREAD

**Discussion by JimjunZhang** [stephenberry/glaze#1423] [Q&A]
Hello,
Is there any limit in Glaze on the number of the fields in a structure to be outputted to json?   In my project, we have a big structure which has ~100 member fields, and the number is still increasing slowly.

**Comment by stephenberry** [comment]
If you use the `glz::meta` approach there is no limit to the number of members. For pure reflection the limit right now is 128 members, but this could be increased if there is a need.

**Reply by JimjunZhang** [reply]
Thanks! It would be great if you can increase it to a bigger number.

**Reply by stephenberry** [reply]
Would 256 be sufficient? It just adds a lot of boilerplate code.

**Reply by JimjunZhang** [reply]
256 should be sufficient.  Thanks!

**Reply by JimjunZhang** [reply]
When do you expect to increase to 256?

**Reply by stephenberry** [reply]
@JimjunZhang, This is going to add about 3,000 lines of code. This could have a compile time effect on users who don't need such a high field count. I think I'm going to create a separate optional header that will support these higher field counts.

If you look at the file "include/glaze/reflection/to_tuple.hpp" you'll see why this header has to be so big.

If you wanted to write a script to generate a header (e.g. `to_tuple_above_128.hpp`) and submit a merge request, I'd be happy to merge it in. Otherwise, I'll try to write a script soon to generate this.

Note that we can use a `requires (N <= 128)` for the current `to_tuple` and `requires (N > 128 && N <= 256)` for the second `to_tuple`, which will make including the `to_tuple_above_128.hpp` automatically increase the supported size to 256 without any other changes.
